libs/s25main/desktops/dskHostGame.cpp

```diff
@@ -1042,8 +1057,23 @@ bool dskHostGame::checkOptions()
     if(forceOptions)
         return true;
     const GlobalGameSettings& ggs = gameLobby->getSettings();
-    if(ggs.getSelection(AddonId::PEACEFULMODE)
-       && (ggs.objective == GO_CONQUER3_4 || ggs.objective == GO_TOTALDOMINATION))
+    if(ggs.objective == GO_ECONOMYMODE && !ggs.getSelection(AddonId::PEACEFULMODE))
+    {
+        WINDOWMANAGER.Show(std::make_unique<iwMsgbox>(
+          _("Economy mode"),
+          _("You chose the economy mode. In economy mode the player or team that collects the most of certain goods "
+            "wins (check the economic progress window in game).\n\nSome players like to play this objective in "
+            "peaceful mode. Would you like to adjust settings for a peaceful "
+            "game?\n"
+            "Choosing Yes will activate peaceful mode, ban catapults and disable buildings receiving coins by default. "
+            "After clicking Yes you will be able to review the changes and then start the game by clicking the Start "
+            "game button again. If you like to play economy mode with attacks, "
+            "choosing No will start the game without any "
+            "changes."),
+          this, MSB_YESNOCANCEL, MSB_QUESTIONGREEN, 10));
+        return false;
+    } else if(ggs.getSelection(AddonId::PEACEFULMODE)
```

Comment:
Maybe better. `ggs.isEnabled(AddonId::PEACEFULMODE)`?
Also can you reformat the message? Line breaks look arbitrary, maybe add one after `\n` and then check that the lines in code are ok? (i.e. the `"game?\n"` is odd)
Besides that the message is pretty verbose but I like it. Maybe I'd omit the  `(check the economic progress window in game)` but I guess that is good for newcomers
I'd also add a newline before `If you like to play economy mode with attacks,` and maybe make it similar to the Yes part: "Choosing No will start the game without any changes." should be enough

Reply:
I like the `(check the economic progress window in game)` so I kept it for now. The rest I changed